### PR TITLE
fix(history-service): strip the quotes around the legacy members_removed account

### PR DESCRIPTION
**Pull request by evelynchang5525**
## Summary

Follow-up to #404, which shipped the legacy `members_removed` display-name resolution against the wrong message format. The stored sentence **quotes** the account:

```
"bob" has been removed from the channel.
```

The merged matcher cut only the trailing suffix and took the whole remaining prefix as the account, so it looked up `"bob"` **with the quote characters included**. That value matches no `users.account` document, so every row fell through to the unresolved-account path and kept its raw text. The feature was a silent no-op on real data — safe, but with no effect.

## The fix

`extractRemovedAccount` now matches the quoted form specifically, returns the account from between the quotes, and the rewrite re-renders the resolved name back inside them:

| | |
|---|---|
| Stored | `"bob" has been removed from the channel.` |
| Returned | `"Bob 鮑勃" has been removed from the channel.` |

Keeping the quotes also matches how `room-worker/sysmsg.go` quotes names on modern `member_removed` messages, so migrated and current rows render alike.

An **unquoted** prefix no longer matches at all. Combined with the `type == "members_removed"` gate, that tightens the guard against rewriting an ordinary user message that happens to end in the same sentence.

## Changes

- **`sysmsgname.go`** — `extractRemovedAccount` requires `"…"` around a non-empty account and returns the inner text; new `quoteRemoved` helper renders the sentence back with the quotes intact.
- **`sysmsgname_test.go`** — matcher table rewritten to the real format, now 14 cases.
- **`messages_test.go`** — the `LoadHistory` end-to-end test uses the quoted shape.
- **`docs/client-api.md`** — the `msg` field now states the actual stored format and that the quotes are preserved.

## Edge cases added

`""` (empty quotes), a lone `"`, opening-quote-only, closing-quote-only, unquoted prefix, and an account containing its own quote (`"bo"b"` → `bo"b`, since only the outermost pair are delimiters).

## Behavior unchanged

Everything else from #404 holds: one batched `$in` per response regardless of how many rows qualify, zero store calls on a page with no qualifying rows, and best-effort degradation — a lookup failure or an account with no user document leaves the row exactly as it reads today.

## Verification

- `make test SERVICE=history-service` — pass, with `-race`
- `make lint` — **0 issues**
- Coverage — **100%** on all four functions in `sysmsgname.go`
- Red phase confirmed the defect before the fix: `expected: "bob"` / `actual: "\"bob\""`

## Note on the branch

#404 was squash-merged, so this branch was brought up to date by merging `main` in rather than stacking on merged history. That merge resolved two conflicts against work landed since: the `closeOnCleanupIn(t, …)` test helper and the preview warm-back config (#406) — both taken from `main`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01HV2ybDmGevhTNb6MpDmQDZ

---
_Generated by [Claude Code](https://claude.ai/code/session_01HV2ybDmGevhTNb6MpDmQDZ)_

### CODE REVIEW

**Comment by julianshen** [comment]
**No actionable comments.** ✅

### Related work

- [PR #404](https://github.com/hmchangw/newchat/pull/404) — feat\(history-service\): resolve legacy members\_removed display names (merged)
- [PR #406](https://github.com/hmchangw/newchat/pull/406) — Queue preview warm-backs off the request path (merged)

<details>
<summary>📒 Files reviewed (4)</summary>

* `docs/client-api.md`
* `history-service/internal/service/messages_test.go`
* `history-service/internal/service/sysmsgname.go`
* `history-service/internal/service/sysmsgname_test.go`

</details>
<details>
<summary>⚙️ Rules run (11)</summary>

* `cassandra`
* `dead-code`
* `distributed-system`
* `elasticsearch`
* `mongodb`
* `nats-large-payload`
* `nats`
* `oom`
* `performance-and-concurrency`
* `silent-failure`
* `tgd-review`

</details>

---

_🤖 Posted by [tGDBot](https://github.com/julianshen/tGDBot) using `openai-codex/gpt-5.6-sol`_

File: history-service/internal/service/sysmsgname.go [hunk 1]
```diff
@@ -21,16 +21,32 @@ const (
...
 	}
-	account, found := strings.CutSuffix(m.Msg, removedFromChannelSuffix)
-	if !found || account == "" {
+	quoted, found := strings.CutSuffix(m.Msg, removedFromChannelSuffix)
+	if !found {
+		return "", false
+	}
+	// Needs both quotes plus at least one character between them; `""` and a
+	// lone `"` name nobody. An inner quote belongs to the account itself.
+	if len(quoted) < 3 || !strings.HasPrefix(quoted, `"`) || !strings.HasSuffix(quoted, `"`) {
```
Comment (julianshen) [inline history-service/internal/service/sysmsgname.go:40]: _🏷️ protocol compatibility_ | _🟠 Warning_ | _⚡ Quick fix_ | _`distributed-system`_

**Unquoted legacy rows are no longer resolved**

The base implementation and tests treated `bob has been removed from the channel.` as a valid legacy row, but the new parser rejects it unless the account is quoted. After deployment, replaying or reading any previously supported unquoted row makes `extractRemovedAccount` return false, so the user lookup is skipped and clients see the stored account identifier instead of the display name. Accept both stored formats during the compatibility window and preserve each row's original quoting style when substituting the name; retain tests for both quoted and unquoted historical rows.

<details>
<summary>🤖 Prompt for AI Agents</summary>

```
In `history-service/internal/service/sysmsgname.go` around line 40: The base implementation and tests treated `bob has been removed from the channel.` as a valid legacy row, but the new parser rejects it unless the account is quoted. After deployment, replaying or reading any previously supported unquoted row makes `extractRemovedAccount` return false, so the user lookup is skipped and clients see the stored account identifier instead of the display name. Accept both stored formats during the compatibility window and preserve each row's original quoting style when substituting the name; retain tests for both quoted and unquoted historical rows.

Fix only this issue, keep the change minimal, and make sure the tests still pass.
```

</details>

---

_🤖 Posted by [tGDBot](https://github.com/julianshen/tGDBot) using `openai-codex/gpt-5.6-sol`_

**Review by julianshen — COMMENTED** [review]
### tGDBot review of `893bae3` — 1 finding in 4 files

1 🟠 warning
Findings not shown inline are in the [review summary](https://github.com/hmchangw/newchat/pull/413#issuecomment-5455425158), which is authoritative for what this run found and where each finding went.
Rules: `cassandra`, `dead-code`, `distributed-system`, `elasticsearch`, `mongodb`, `nats-large-payload`, `nats`, `oom` +3 more

<details>
<summary>How to read an inline comment</summary>

- Each one opens with `category | severity | effort | rule`.
- Severity is one of 🔴 Blocking · 🟠 Warning · 🔵 Suggestion.
- Effort is ⚡ Quick fix or 🏗️ Heavy lift — the SIZE of the change, not its importance.
- The collapsed 🤖 block is a ready-made prompt for handing that one finding to a coding agent.

</details>

<details>
<summary>Replying to @julianshen</summary>

In a finding's own thread: `@julianshen explain`, `@julianshen reconsider <reason>`.

Anywhere on the pull request: `@julianshen check latest`, `@julianshen review focus: <direction>`,
`@julianshen remember <lesson>`, `@julianshen memories`, `@julianshen forget <memory-id>`.

One command per comment.

</details>

---

_🤖 Posted by [tGDBot](https://github.com/julianshen/tGDBot) using `openai-codex/gpt-5.6-sol`_

**Review by mliu33 — APPROVED** [review]

File: history-service/internal/service/sysmsgname.go (hunk 1 above)
Comment (evelynchang5525) [reply history-service/internal/service/sysmsgname.go:40]: Verified the mechanics — accurate: an unquoted row that #404's matcher accepted is rejected here. But the "previously supported" framing doesn't hold against what that code actually did in production, so I'm leaving the behaviour as-is and flagging it to the author rather than adding the compatibility branch.

Two things:

**#404 resolved nothing in production.** Its matcher cut only the trailing suffix and took the whole prefix as the account, so on the real stored form it extracted `"bob"` *including the quote characters*. That value matches no `users.account` document, so every row fell through to the unresolved-account path and kept its raw text. The unquoted branch was reachable only for a format that, per the repo owner, is not what these rows store — the sample the original implementation was written against was later corrected to the quoted form. So nothing that worked before stops working: before #404 there was no resolution, under #404 there was no resolution, and here quoted rows resolve.

**The strictness is load-bearing, not incidental.** `members_removed` rows are migrated data; the `type` gate is the primary guard, and requiring the quotes is the second one that keeps an ordinary user message ending in the same sentence from being rewritten. Accepting the bare form would re-open that and add a branch for data we have no evidence exists.

If mixed formats *do* exist in the migrated corpus, this is worth doing and I'll add it — dual-form matching plus preserving each row's original quoting on substitution, with tests for both. That's a question about the data rather than the diff, so I've put it to the author and am leaving this thread open for them to decide.

---
_Generated by [Claude Code](https://claude.ai/code)_
